File: AtCoder/Browns/cpp/AGC032-B-Balanced_Neighbors.cpp
#include <bits/stdc++.h>
#define MAX_N 100
using namespace std;
typedef long long ll;

ll N;
ll d[MAX_N][MAX_N];

int main() {
  cin >> N;
  ll cnt = 0;
  for (ll i = 1; i < N; ++i) {
    for (ll j = i + 1; j < N + 1; ++j) {
      if (N % 2 != 0 && i + j == N) continue;
      if (N % 2 == 0 && i + j == N + 1) continue;
      d[i - 1][j - 1] = true;
      ++cnt;
    }
  }
  printf("%lld\n", cnt);
  for (ll i = 0; i < N - 1; ++i) {
    for (ll j = i + 1; j < N; ++j) {
      if (d[i][j]) printf("%lld %lld\n", i + 1, j + 1);
    }
  }
}

File: AtCoder/Browns/python/AGC032-B-Balanced_Neighbors.py
N = int(input())
d = [[False] * N for _ in range(N)]
cnt = 0
for i in range(1, N):
    for j in range(i + 1, N + 1):
        if N % 2 != 0 and i + j == N:
            continue
        if N % 2 == 0 and i + j == N + 1:
            continue
        d[i - 1][j - 1] = True
        cnt += 1
print(cnt)
for i in range(N - 1):
    for j in range(i + 1, N):
        if d[i][j]:
            print(i + 1, j + 1)
